### ISSUE TITLE
PDF is not generated in production ENV

### ISSUE THREAD

**Issue by lucasloami** [mileszs/wicked_pdf#403]
Hi, 

I have a Rails app that uses wicked_pdf gem and it works pretty well in my development environment. However, when I deploy the app in my server (it's in a Ubuntu VM in Azure and it uses unicorn and nginx) it doesn't work. I researched a lot in Google but I was not able to find any solution that worked. 

Here is what I have in my server:

OS: Ubuntu 14.04 (it's an MS Azure VM)
Server: all the required libraries are installed - ruby, nginx, unicorn, etc.
wkhtmltopdf: the binary file is installed. I downloaded it from [this website](http://wkhtmltopdf.org/) the Linux (Ubuntu Trusty) version (Ubuntu 10.04.1). Its installed in `/usr/local/bin`

My initializer is configured like this

```ruby
WickedPdf.config = {exe_path: '/usr/local/bin/wkhtmltopdf'}
```

If I run the command in my console, it gives to me a correct pdf file. So this means that the library is working in the server. The point is: for some reason, the wicked_pdf gem cannot use it

```console
wkhtmltopdf http://google.com google.pdf
```

I followed all the instructions mentioned in the README.md, but nothing is working.

Can you help me?

**Comment by unixmonkey** [comment]
Is there an error message or something like that?

**Comment by lalitlogical** [comment]
I also have same problem on production server for a rails app which is running on sub uri. In development environment, it run like charm. But on production server, only render the layout.

Production:
OS: Ubuntu 14.04 on digital ocean
Server: nginx
Rails server: puma

Local:
OS: OS X Yosemite (10.10)
Server: nginx 
Rails server: puma

I have replicated the production server on my local system. Means I am running my rails app under nginx server with sub uri. All things work perfect. I have installed all things on server. Actually it is running perfect before sub uri. But I have provided the new absolute urls with sub uri for all javascript, stylesheets and images. Also it is showing perfect html which I requires from it when I am using :show_as_html => params[:debug].present?

Actually when html load, I use javascript methods to fill some dynamic html content to existing divs. Its fill perfectly on development environments but not on production server. It may be not found my external javascript file but why it is working on development?

I have made every possible changes for it. I have not found any error on production server related to it.

Can you please guide me for the same issue?

**Comment by lucasloami** [comment]
Unfortunatelly not. I checked all the possible logs in the server: nginx logs, rails app production.log and nothing seems to be put there. 

I can take look one more time to try to find something.

**Comment by lucasloami** [comment]
Is there some speficic configuration to be done in the production server in order to make this gem works, @unixmonkey ?

**Comment by lucasloami** [comment]
Does wicked_pdf create some temp PDF file before generate the final one?

**Comment by unixmonkey** [comment]
@lucasloami

No, but it does generate a HTML tempfile of the rendered page, and passes that into wkhtmltopdf

**Comment by lalitlogical** [comment]
I have been resolved my issue. Please refer this https://github.com/mileszs/wicked_pdf/issues/344

**Comment by mariusbutuc** [comment]
@unixmonkey this issue can likely be closed in favour of #344.
